nemo/core/neural_modules.py

```diff
@@ -39,6 +38,15 @@
 YAML = YAML(typ='safe')
 
 
+class ModuleType(Enum):
+    """ Back-end independent module types """
+
+    module = 0
```

Comment:
can you please this set to: "trainable, nontrainable, datalayer, loss" ?

Reply:
Good catch... but once again: I did that for a reason. I truly believe that we should get rid of a NonTrainableNM class. The only thing it is doing now it is stopping the gradient backpropagation - and **this is simply wrong**. 

Anyway, this is a subject for a different conversation. Added "nontrainable" enum.